test/test_links.cpp

```diff
@@ -1605,4 +1605,49 @@ TEST(Links_DetachedAccessor)
     CHECK_NOT(link_list.is_attached());
 }
 
+TEST(Unresolved_Mixed_links)
+{
+    Group g;
+
+    auto source_table = g.add_table_with_primary_key("source", type_Int, "source_id");
+    auto dest_table = g.add_table_with_primary_key("dest", type_Int, "dest_id");
+    auto list_mixed_col = source_table->add_column_list(type_Mixed, "list of mixed");
+    auto set_of_mixed_col = source_table->add_column_set(type_Mixed, "set of mixed");
+
+    auto source_obj = source_table->create_object_with_primary_key(0);
+    auto dest_obj = dest_table->create_object_with_primary_key(1);
+
+    Lst<Mixed> list_mixed = source_obj.get_list<Mixed>(list_mixed_col);
+    Set<Mixed> set_mixed = source_obj.get_set<Mixed>(set_of_mixed_col);
+
+    list_mixed.add(ObjLink{dest_table->get_key(), dest_obj.get_key()});
+    set_mixed.insert(ObjLink{dest_table->get_key(), dest_obj.get_key()});
+
+    size_t expected_size = 1;
+    CHECK_EQUAL(list_mixed.size(), expected_size);
+    CHECK_EQUAL(set_mixed.size(), expected_size);
+
+    dest_obj.invalidate(); // send to graveyard, and transform all incoming links to be unresolved
+
+    // difference between list/set of links and lst/set of Mixed, size returns the whole list of links also the
+    // null/unresolved ones
+    expected_size = 1;
+    CHECK_EQUAL(list_mixed.size(), expected_size); // fails
+    CHECK_EQUAL(set_mixed.size(), expected_size);  // fails
+
+    auto check_mixed_link = [&](Mixed link) {
+        auto is_null = link.is_null();
+        auto is_unresolved = link.is_unresolved_link();
+        CHECK(is_null || is_unresolved);
+    };
+    if (list_mixed.size()) {
+        auto link = list_mixed.get(0);
+        check_mixed_link(link);
```

Comment:
I think the check here should just be that the link is null.

Reply:
There is a difference between a mixed link that is null or a link that is unresolved. Do you want to return always a null Mixed for an unresolved link?

Reply:
Yes, this is what we do in Dictionary

Reply:
No longer applies to the current code